source/features/delete-review-comments-faster.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import delegate from 'delegate-it';
+import {observe} from 'selector-observer';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+
+function deleteComment(comment: HTMLElement): void {
+	select('.dropdown-menu .js-comment-delete > button', comment)!.click();
+}
+
+function onButtonClick(event: delegate.Event): void {
+	const comment = (event.target as HTMLElement).closest<HTMLElement>('.js-comment')!;
+	const dropdownFragment = select('include-fragment.SelectMenu-loading', comment);
+	if (dropdownFragment) {
+		select('.timeline-comment-actions > details:last-of-type', comment)!.dispatchEvent(new Event('mouseover'));
```

Comment:
I think we have a helper function to do exactly this

Reply:
I thought so but couldn't find one. Are you referring to this maybe? https://github.com/sindresorhus/refined-github/pull/3749#discussion_r532286271

Reply:
Yes

Reply:
We could extract the logic into a more generic helper and use it in both features.

Reply:
If it's similar enough yes. If it requires conditions probably not

Reply:
Alternatively, can you check if we can use _any_ authenticity token to delete it? This way we can just replicate the form without having to wait for it.

Note: The form should still probably be created inline and managed by GitHub so we get:

- native error handling
- native prompt, which I think it's missing altogether here and that's not ok

Reply:
> native prompt, which I think it's missing altogether here and that's not ok

I'm seeing a confirmation dialog when deleting though?

Reply:
True. The rest of the review still applies though

Reply:
I checked. The delete form’s authenticity token cannot be replaced with another token, so we can't recreate the form manually.